diff -urN Shortwave-orig/src/audio/player.rs Shortwave/src/audio/player.rs
--- Shortwave-orig/src/audio/player.rs	2020-10-09 16:40:26.302135726 +0300
+++ Shortwave/src/audio/player.rs	2020-10-09 16:56:47.861256645 +0300
@@ -389,9 +389,12 @@
             // Remove unsupported characters from the file name
             let title_raw = utils::simplify_string(title.to_string());
             // Limit file name to 200 chars
-            let title_vec = title_raw.chars().collect::<Vec<_>>();
-            let cut_to_length = title_vec.len().min(200);
-            let title = title_vec[..cut_to_length].iter().cloned().collect::<String>();
+            let mut limit = 200;
+                while (title_raw.len() > 200) && (!title_raw.is_char_boundary(limit)) {
+                    limit -= 1;
+                }
+            let title = &title_raw[..limit.min(title_raw.len())];
+
 
             let mut path = path::CACHE.clone();
             path.push("recording");
